ONSdigital/response-operations-ui
# Ce event error message update

# Motivation and Context

The create and update collection exercise event forms were not giving informative errors on failure, causing the user to sometimes not know why the submission had failed.

# What has changed

Added better handling of how errors are displayed to the user
also a minor fix of making the cancel button work as expected rather than acting as a back(history) button.


added some unit tests for testing error 'categories' as messages are now being flashed asserting for specific messages will need to be done through acceptance tests.
# How to test?

Run ResOps unittests
manually enter invalid information to the create/update CE event forms in the surveys tab and check you are being given an informative error message.



# Links

https://trello.com/c/tVq25lkO/791-fix-error-messaging-when-validating-the-ce-event-create-update-forms


Collex PR: https://github.com/ONSdigital/rm-collection-exercise-service/pull/143

# Screenshots (if appropriate):
